federation/pkg/kubefed/init/init.go

```diff
@@ -586,6 +601,28 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 	return clientset.Extensions().Deployments(namespace).Create(dep)
 }
 
+func createFedConfigMap(config util.AdminConfig, fedName, kubeconfig, dnsZoneName string, dryRun bool) (*v1.ConfigMap, error) {
+	fedClientSet, err := config.FederationClientset(fedName, kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	configMap := &v1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "kube-dns",
+			Namespace: "kube-system",
+		},
+		Data: map[string]string{
+			"federations": fmt.Sprintf("%s=%s", fedName, dnsZoneName),
```

Comment:
This clobbers the existing `federations'` value. Also, create below fails if the ConfigMap already exists. I think you should  update or create depending on the existence of this ConfigMap.